File: sample/python/payment.py
"""
Payment processing module
"""

def process_payment(amount, user_id):
    """
    Process payment for a user
    """
    result = charge_credit_card(amount, user_id)
    
    update_user_balance(user_id, amount)
    
    return result

def charge_credit_card(amount, user_id):
    if amount <= 0:
        raise ValueError("Amount must be positive")
    
    return {"status": "success", "transaction_id": "12345"}

def update_user_balance(user_id, amount):
    pass

File: sample/java/Payment.java
package com.example.app;

/**
 * Payment processing service
 */
public class PaymentProcessor {
    
    public boolean processPayment(double amount, String userId) {
        validateAmount(amount);
        
        boolean result = chargeCard(amount, userId);
        
        updateUserBalance(userId, amount);
        
        return result;
    }
    
    private void validateAmount(double amount) {
        if (amount <= 0) {
            throw new IllegalArgumentException("Amount must be positive");
        }
    }
    
    private boolean chargeCard(double amount, String userId) {
        // Simulated payment processing
        return true;
    }
    
    private void updateUserBalance(String userId, double amount) {
        
    }
}
